test_conformance/half/Test_vStoreHalf.cpp

```diff
@@ -755,556 +915,771 @@ int Test_vStoreHalf_private( cl_device_id device, f2h referenceFunc, d2h doubleR
 
-        error = clEnqueueWriteBuffer(gQueue, gInBuffer_single, CL_FALSE, 0, count * sizeof(float ), gIn_single, 0, NULL, NULL);
-        if (error) {
-            vlog_error( "Failure in clWriteBuffer\n" );
+        error = clEnqueueWriteBuffer(gQueue, gInBuffer_single, CL_FALSE, 0,
+                                     count * sizeof(float), gIn_single, 0, NULL,
+                                     NULL);
+        if (error)
+        {
+            vlog_error("Failure in clWriteBuffer\n");
             gFailCount++;
             goto exit;
         }
 
-        if (gTestDouble) {
+        if (gTestDouble)
+        {
             ThreadPool_Do(ReferenceD, threadCount, &dref);
 
-            error = clEnqueueWriteBuffer(gQueue, gInBuffer_double, CL_FALSE, 0, count * sizeof(double ), gIn_double, 0, NULL, NULL);
-            if (error) {
-                vlog_error( "Failure in clWriteBuffer\n" );
+            error = clEnqueueWriteBuffer(gQueue, gInBuffer_double, CL_FALSE, 0,
+                                         count * sizeof(double), gIn_double, 0,
+                                         NULL, NULL);
+            if (error)
+            {
+                vlog_error("Failure in clWriteBuffer\n");
                 gFailCount++;
                 goto exit;
             }
         }
 
-        for (vectorSize = kMinVectorSize; vectorSize < kLastVectorSizeToTest; vectorSize++) {
+        for (vectorSize = kMinVectorSize; vectorSize < kLastVectorSizeToTest;
+             vectorSize++)
+        {
             // Loop through vector sizes
             fchk.vsz = g_arrVecSizes[vectorSize];
             dchk.vsz = g_arrVecSizes[vectorSize];
 
-            for ( addressSpace = 0; addressSpace < 3; addressSpace++) {
+            for (addressSpace = 0; addressSpace < 3; addressSpace++)
+            {
                 // Loop over address spaces
                 fchk.aspace = addressSpaceNames[addressSpace];
                 dchk.aspace = addressSpaceNames[addressSpace];
 
-                cl_uint pattern = 0xdeaddead;
-                memset_pattern4( gOut_half, &pattern, BUFFER_SIZE/2);
-
-                error = clEnqueueWriteBuffer(gQueue, gOutBuffer_half, CL_FALSE,
-                                             0, count * sizeof(cl_half),
-                                             gOut_half, 0, NULL, NULL);
-                if (error) {
-                    vlog_error( "Failure in clWriteArray\n" );
+                error = RunKernel(device, resetKernel, gInBuffer_single,
+                                  gOutBuffer_half, count, 0);
+                if (error)
+                {
                     gFailCount++;
                     goto exit;
                 }
 
-                error = RunKernel(device, kernels[vectorSize][addressSpace], gInBuffer_single, gOutBuffer_half,
-                                       numVecs(count, vectorSize, aligned) ,
+                memset_pattern4(gOut_half, &pattern, BUFFER_SIZE / 2);
```

Comment:
This may not matter much in the grand scheme of things, but gOut_half is only used as a destination for the call to clEnqueueReadBuffer below now, so memset'ing the pattern each time before calling clEnqueueReadBuffer isn't necessary.

In order, I think my preference would be:

1. Remove it completely because it isn't needed.
2. If we can't remove it completely, move it immediately before the call to clEnqueueReadBuffer, perhaps with a comment why it is there.
3. If we don't want to remove it or move it, I suppose we can keep it as-is, because it's functionally correct now even if it is confusing.

Reply:
Removed.